src/server/bg_services/log_replication_scanner.js

```diff
@@ -136,45 +143,41 @@ class LogReplicationScanner {
         return { copy_keys, delete_keys };
     }
 
-    async process_candidates_sync_version(src_bucket, dst_bucket, candidates) {
+    async process_candidates_sync_version(src_bucket, dst_bucket, candidates, sync_deletions) {
         const bucketDiff = new BucketDiff({
             first_bucket: src_bucket.name,
             second_bucket: dst_bucket.name,
             version: true,
             connection: this.noobaa_connection,
-            for_replication: config.BUCKET_DIFF_FOR_REPLICATION
+            for_replication: config.BUCKET_DIFF_FOR_REPLICATION,
+            for_deletion: sync_deletions
         });
-
-        let copy_keys = {};
-        // TODO: support delete flow in object versions replication
-        // const delete_keys = [];
+        // diff_keys can be either copy_keys or delete_keys based on the sync_deletions
+        let diff_keys = {};
 
         for (const candidate of Object.values(candidates)) {
             const action = candidate.action;
             if (action === 'skip') {
                 // Log skipped candidate key
-                dbg.log1('process_candidates: skipped_key: ', candidates.key);
+                dbg.log1('process_candidates: skipped_key: ', candidate.key);
             } else {
                 // The action should not matter to get_buckets_diff, we will evaluate the action inside.
                 // This is because the order of the versions matter.  hance we just need the hint from the logs
                 // regarding which key was touched
                 const { keys_diff_map } = await bucketDiff.get_buckets_diff({
-                    prefix: candidates.key,
+                    prefix: candidate.key,
                     max_keys: Number(process.env.REPLICATION_MAX_KEYS) || 1000, //max_keys refers her to the max number of versions (+deletes)
                     current_first_bucket_cont_token: '',
                     current_second_bucket_cont_token: '',
                 });
-                // Currently, as get_buckets_diff is not supporting deletions, we will just pass the keys_diff_map as copy
-                // This needs to be reevaluated once the delete is supported.
-                copy_keys = { ...copy_keys, ...keys_diff_map };
+
+                diff_keys = { ...diff_keys, ...keys_diff_map };
             }
         }
 
-        dbg.log1('process_candidates_sync_version:: copy_keys', copy_keys);
-        //TODO support delete flow and return also the delete
-        // returning copy_keys and delete_keys after processing candidates
-        // return { copy_keys, delete_keys };
-        return { copy_keys, delete_keys: [] };
+        dbg.log1('process_candidates_sync_version:: diff_keys', diff_keys);
```

Comment:
We might want to add here the value of sync_deletions

Reply:
Sure, will add that.